assets/js/pages/BattleDetails.js

```diff
@@ -124,19 +126,16 @@ function TeamDetails(props) {
 class BattleDetails extends React.Component {
   constructor(props) {
     super(props);
-
     this.state = {
-      battle: null,
+      battleId: props.match.params.pk,
```

Comment:
It's not a good idea to put the `pk` in the component's `state`, since it's a value that comes from a prop  - and that's not what the state should be used for, in most cases (this could force you to write a `componentWillReceiveProps` method to pass down any new prop values to the state).

You can use something like `const { params } = props.match` and access the `pk` from it instead.